backend/api/Controllers/AreaController.cs

```diff
@@ -0,0 +1,248 @@
+                _logger.LogInformation(
+                    "Succesfully created new area with id '{areaId}'",
+                    newArea.Id
+                );
+                return CreatedAtAction(
+                    nameof(GetAreaById),
+                    new { id = newArea.Id },
+                    newArea
+                );
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error while creating new area");
+                throw;
+            }
+        }
+
+
+        /// <summary>
+        /// Add a safe position to a area
+        /// </summary>
+        /// <remarks>
+        /// <para> This query adds a new safe position to the database </para>
+        /// </remarks>
+        [HttpPost]
+        [Authorize(Roles = Role.Admin)]
+        [Route("{asset}/{installationName}/{deckName}/{areaName}/safe-position")]
+        [ProducesResponseType(typeof(Area), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<Area>> AddSafePosition(
+            [FromRoute] string asset,
+            [FromRoute] string installationName,
+            [FromRoute] string deckName,
+            [FromRoute] string areaName,
+            [FromBody] Pose safePosition
+        )
+        {
+            _logger.LogInformation("Adding new safe position");
+            try
+            {
+                var area = await _areaService.AddSafePosition(asset, areaName, new SafePosition(safePosition));
+                if (area != null)
+                {
+                    _logger.LogInformation("Succesfully added new safe position for asset '{assetId}' and name '{name}'", asset, areaName);
+                    return CreatedAtAction(nameof(GetAreaById), new { id = area.Id }, area); ;
+                }
+                else
+                {
+                    _logger.LogInformation("Creating Area for asset '{assetId}' and name '{name}'", asset, areaName);
+                    // Cloning to avoid tracking same object
+                    var tempPose = ObjectCopier.Clone(safePosition);
+                    area = await _areaService.Create(
+                        new CreateAreaQuery
+                        {
+                            AssetCode = asset,
+                            AreaName = areaName,
+                            InstallationName = installationName,
+                            DeckName = deckName,
+                            DefaultLocalizationPose = new Pose()
+                        },
+                        new List<Pose> { tempPose }
+                    );
+                    return CreatedAtAction(nameof(GetAreaById), new { id = area.Id }, area);
+                }
```

Comment:
I would return a NotFound(area) instead of creating a new area. Creating a new area should be done through the dedicated endpoint and not as a side effect where one might not expect it. For example, if someone accidentely wrote the area name incorrectly we would suddenly get a new area in the database.